CONTRIBUTING.md

```diff
@@ -1,183 +1,71 @@
 
-      self.attr5 = None
-      """str: Docstring *after* attribute, with type specified."""
+3. Create a branch for the contribution. You should create a branch from the GitHub issue page, by clicking the `Create a branch` button under the `Development` section on the right;
 
-```
+4. Start implementing your changes and [commit](https://github.com/git-guides/git-commit) them;
 
-An example how to describe a function:
+5. Once the changes are mature enough to be discussed, create a `draft` [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) in the repository;
 
-```bash
-def function_with_types_in_docstring(param1, param2):
-  """Example function with types documented in the docstring.
+6. When you feel like your changes are ready for production, mark the pull request as `ready`;
 
-  `PEP 484`_ type annotations are supported. If attribute, parameter, and
-  return types are annotated according to `PEP 484`_, they do not need to be
-  included in the docstring:
+7. Wait for [review](#reviewing-process);
 
-  Args:
-      param1 (int): The first parameter.
-      param2 (str): The second parameter.
+8. Once a reviewer has accepted your pull request, you are in charge of [merging](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/merging-a-pull-request). We use `squash and merge` to keep history clear and simple.
 
-  Returns:
-      bool: The return value. True for success, False otherwise.
+### Reviewing process
 
-  .. _PEP 484:
-      https://www.python.org/dev/peps/pep-0484/
+Pull requests marked as ready have to be reviewed before they can be merged.
 
-  """
+**The reviewer must be, if possible, from a different facility than the author.**
 
-```
+The reviewer is in charge of verifying the following conditions:
 
-You can use [autopep8](https://pypi.org/project/autopep8/) and [black](https://pypi.org/project/autopep8/) to format your code:
+-   the contribution matches the issue requirements
+-   no breaking change has been introduced without proper discussion
+-   sufficient testing has been implemented
+-   CI checks are green
+-   the coding style is respected
 
-```bash
-autopep8 -a -r -j 0 -i --max-line-length 88 ./
-black --safe ./
-```
+If necessary, make comments on the code with clear hints on what to do for the author.
 
-### Continuous integration (CI)
+When the code is ready for production, mark the pull request as ready.
 
-For continuous integration Gitlab-CI is used.
+**When the reviewer validates the pull request, its author is in charge of merging.**
 
-### Additional notes
+### Coding style guidelines
 
-Issue and Pull request Labels
+It is very important to write clean and readable code. Therefore we follow the [PEP8 guidlines](https://www.python.org/dev/peps/pep-0008/).
```

Comment:
May I propose to use instead Black (https://black.readthedocs.io/en/stable/index.html).
This is an autoformatting program (does the work for you), a rigid subset of PEP8 (well standardised), but uses a slightly longer line length (88 characters), leading to fewer lines.

Reply:
Black is already used an enforced in CI. We should probably mention that :)

Reply:
updated